application/Common/HookLibrary/dllmain.cpp

```diff
@@ -19,6 +22,8 @@ LRESULT CALLBACK GetMsgProc(int nCode, WPARAM wParam, LPARAM lParam)
 //this is the hook procedure
 {
     LRESULT retVal = CallNextHookEx(GetMessageHook, nCode, wParam, lParam);
+    if (!mappedFileHandle)
+        MapSharedMemory();
```

Comment:
I will add another commit with error handling here (if mapping fails, i need to return immediately or bad things will happen)